tools/secret-rotation/Azure.Sdk.Tools.SecretRotation.Core/RotationPlan.cs

```diff
@@ -134,51 +134,67 @@ private async Task RotateAsync(string operationId, SecretState currentState, boo
         DateTimeOffset invocationTime = this.timeProvider.GetCurrentDateTimeOffset();
 
         SecretValue newValue = await OriginateNewValueAsync(operationId, invocationTime, currentState, whatIf);
+        // TODO: some providers will issue secrets for longer than we requested. Should we propagate the real expiration date, or the desired expiration date?
+
+        await WriteValueToPrePrimaryStoresAsync(newValue, currentState, whatIf);
 
-        await WriteValueToSecondaryStoresAsync(newValue, currentState, whatIf);
+        if (OriginStore != PrimaryStore)
+        {
+            await WriteValueToPrimaryAsync(newValue, currentState, invocationTime, whatIf);
+        }
 
-        await WriteValueToPrimaryAndOriginAsync(newValue, invocationTime, currentState, whatIf);
+        await WriteValueToPostPrimaryStoresAsync(newValue, currentState, whatIf);
+
+        await MarkRotationCompleteAsync(newValue, currentState, invocationTime, whatIf);
```

Comment:
Unrelated to this PR, but would it be easier to set whatIf on the class vs. having to allow for it in all method signatures and remember to propagate?

Reply:
could do.   I was intentionally avoiding putting state on the class, but that doesn't really make sense as it's functionally a stateful rotation context.